packages/python/src/armonik/worker/worker.py

```diff
@@ -68,9 +73,119 @@ def Process(self, request: ProcessRequest, context) -> Union[ProcessReply, None]
-                f"Failed task {''.join(traceback.format_exception(type(e) ,e, e.__traceback__))}",
+                f"Failed task {''.join(traceback.format_exception(type(e), e, e.__traceback__))}",
                 exc_info=e,
             )
 
     def HealthCheck(self, request: Empty, context) -> HealthCheckReply:
         return HealthCheckReply(status=self.health_check())
+
+
+_NOT_SET = object()
+
+
+class ArmoniKWorkerWrapper:
+    def __init__(
+        self,
+        *,
+        processor: Callable[[TaskHandler], Output],
+        logger: Optional[Logger] = None,
+        worker_endpoint: Optional[str] = None,
+        agent_endpoint: Optional[str] = None,
+        channel_options: Optional[Tuple[Tuple[str, str]]] = _NOT_SET,
+    ):
+        if logger is None:
+            ClefLogger.setup_logging(logging.INFO)
+            logger = ClefLogger.getLogger("ArmoniKWorker")
+        if worker_endpoint is None:
+            worker_scheme = (
+                "unix://"
+                if os.getenv("ComputePlane__WorkerChannel__SocketType", "unixdomainsocket")
+                == "unixdomainsocket"
+                else "http://"
+            )
+            worker_endpoint = worker_scheme + os.getenv(
+                "ComputePlane__WorkerChannel__Address", "/cache/armonik_worker.sock"
+            )
+        if agent_endpoint is None:
+            agent_scheme = (
+                "unix://"
+                if os.getenv("ComputePlane__AgentChannel__SocketType", "unixdomainsocket")
+                == "unixdomainsocket"
+                else "http://"
+            )
+            agent_endpoint = agent_scheme + os.getenv(
+                "ComputePlane__AgentChannel__Address", "/cache/armonik_agent.sock"
+            )
+        if channel_options is _NOT_SET:
+            channel_options = (("grpc.default_authority", "localhost"),)
+        self.logger = logger
+        self.worker_endpoint = worker_endpoint
+        self.agent_endpoint = agent_endpoint
+        self.channel_options = channel_options
+        self.processor = processor
+
+    def __call__(self, channel: Optional[Channel] = None):
+        # Start worker
+        self.logger.info("Worker Started")
+        # Use options to fix Unix socket connection on localhost (cf: <GitHub>)
+        if channel is None:
+            channel_ = create_channel(self.agent_endpoint, options=self.channel_options).__enter__()
+        else:
+            channel_ = channel
+        try:
+            worker = ArmoniKWorker(channel_, self.processor, logger=self.logger)
+            self.logger.info("Worker Connected")
+            worker.start(self.worker_endpoint)
+        except Exception as e:
+            if channel is None:
+                channel_.__exit__(type(e), e, e.__traceback__)
+                channel_ = None
+            raise
+        finally:
+            if channel is None and channel_ is not None:
+                channel_.__exit__(None, None, None)
```

Comment:
I am not sure why it could not be a `with` block

Reply:
An external channel should not be reopened and closed inside, as its lifetime is handled outside the function